pkg/middleware/stored_session.go

```diff
@@ -89,6 +99,25 @@ func (s *storedSessionLoader) loadSession(next http.Handler) http.Handler {
 // getValidatedSession is responsible for loading a session and making sure
 // that is is valid.
 func (s *storedSessionLoader) getValidatedSession(rw http.ResponseWriter, req *http.Request) (*sessionsapi.SessionState, error) {
+	var session *sessionsapi.SessionState
+	var err error
+	failed := false
+	attempts := 1 + s.retryAttempts
+	for i := 0; i < attempts; i++ {
+		session, err = s.tryToGetValidatedSession(rw, req)
+		if err == nil {
+			if failed {
+				logger.Printf("Previously invalid session is now valid due to retry. (Retries: %d) \n", i)
```

Comment:
This feels weird to log if we weren't logging the failure refresh cases before this (e.g. how long was my session invalid if I now get a log telling me I had an invalid session?).

Reply:
You're right, this isn't really relevant to the end user, but I think it would be helpful as debug output.

Reply:
I meant I more expected logs of the failures and retries (then eventual success) rather than just getting a log message saying I had an invalid session out of the blue when we finally succeed.